ssl/ssl_locl.h

```diff
@@ -584,6 +592,13 @@ struct ssl_session_st {
         /* The ALPN protocol selected for this session */
         unsigned char *alpn_selected;
         size_t alpn_selected_len;
+        /*
+         * Maximum Fragment Length as per RFC 4366.
+         * If this value does not contain RFC 4366 allowed values (1-4) then
+         * either the Maximum Fragment Length Negotiation failed or was not
+         * performed at all.
+         */
+        unsigned char max_fragment_length;
```

Comment:
You've added this to SSL_SESSION, and the spec says "The negotiated length applies for the duration of the session including session resumptions.", but you have not added this field to the session serialization code in ssl_asn1.c. So if the session gets serialized before we resume it then this value will be lost.